gobblin-modules/gobblin-kafka-common/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSource.java

```diff
@@ -145,6 +154,8 @@ private void setLimiterReportKeyListToWorkUnits(List<WorkUnit> workUnits, List<S
 
   @Override
   public List<WorkUnit> getWorkunits(SourceState state) {
+    this.metricContext = Instrumented.getMetricContext(state, KafkaSource.class);
+
```

Comment:
What about timing isTopicQualified() to get an idea on the cost of the schema fetch?